lib/slather/project.rb

```diff
@@ -51,7 +60,60 @@ def coverage_files
         dedupe(coverage_files)
       end
     end
-    private :coverage_files
+    private :gcov_coverage_files
+
+    def profdata_coverage_files
+      files = profdata_llvm_cov_output.split("\n\n")
+
+      files.map do |source|
+        coverage_file = coverage_file_class.new(self, source)
+        !coverage_file.ignored? ? coverage_file : nil
+      end.compact
+    end
+    private :profdata_coverage_files
+
+    def profdata_coverage_dir
+      if self.scheme
+        Dir["#{build_directory}/**/CodeCoverage/#{self.scheme}"].first
+      else
+        Dir["#{build_directory}/**/#{self.products.first.name}"].first
+      end
+    end
+
+    def binary_file
+      xctest_bundle_file = Dir["#{profdata_coverage_dir}/**/*.xctest"].first
+      if xctest_bundle_file == nil
+        raise StandardError, "No product binary found in #{profdata_coverage_dir}"
+      end
+
+      # Find the matching .app, if any
+      xctest_bundle_file_directory = Pathname.new(xctest_bundle_file).dirname
+
+      app_bundle_file = Dir["#{xctest_bundle_file_directory}/*.app"].first
+      framework_bundle_file = Dir["#{xctest_bundle_file_directory}/*.framework"].first
+      if app_bundle_file != nil
+        app_bundle_file_name_noext = Pathname.new(app_bundle_file).basename.to_s.gsub(".app", "")
+        "#{app_bundle_file}/#{app_bundle_file_name_noext}"
+      elsif framework_bundle_file != nil
+          framework_bundle_file_name_noext = Pathname.new(framework_bundle_file).basename.to_s.gsub(".framework", "")
+          "#{framework_bundle_file}/#{framework_bundle_file_name_noext}"
+      else
+        xctest_bundle_file_name_noext = Pathname.new(xctest_bundle_file).basename.to_s.gsub(".xctest", "")
+        "#{xctest_bundle_file}/#{xctest_bundle_file_name_noext}"
+      end
+    end
+    private :binary_file
+
+    def profdata_llvm_cov_output
+      profdata_coverage_dir = self.profdata_coverage_dir
+      if profdata_coverage_dir == nil || (coverage_profdata = Dir["#{profdata_coverage_dir}/**/Coverage.profdata"].first) == nil
+        raise StandardError, "No Coverage.profdata files found. Please make sure the \"Code Coverage\" checkbox is enabled in your scheme's Test action or the build_directory property is set."
+      end
+      xcode_path = `xcode-select -p`.strip
+      llvm_cov_command = File.join(xcode_path, "Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-cov show -instr-profile #{coverage_profdata} #{binary_file}")
+      `#{llvm_cov_command}`
```

Comment:
@stevepeak, @bgerstle: `.profdata` files are only generated by Xcode if you check that "Gather coverage data". If you don't, you won't even have the Coverage computed, even inside the Xcode GUI.

So even if it's interesting to mention, that's not directly related to `slather coverage` itself. Even if you want the Coverage report inside Xcode itself under the Test Reports tab, you'll need to enable that anyway, even for people not using `slather`.

But indeed it could be interesting to make `slather setup` enforce this setting if it's not the case already (I haven't checked that yet).